### PR TITLE
Update storybook to enable noImplicitAny use

### PR DESCRIPTION
**Pull request by edwardlee-msft**
# What

Storybook snippets to enable noImplictAny use.

Important notes found in this PR:
* We have agreed to only update typescript snippets files as updating all the tsx files in storybook would be extensive and outweigh costs vs benefits.
* One react-components file updated: TagsSurvey.tsx. Updated to include typing required. Due to typescript limitations with the Array function filter, as casting was needed. More information can be found here: https://github.com/microsoft/TypeScript/issues/44373
* One react-composites file updated: CallWithChatBackedCallAdapter.ts. Changed how typing was laid as previous version causes a typescript issue. More information can be found here More information can be found here: https://github.com/microsoft/TypeScript/issues/44373

### CODE REVIEW

File: packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx [hunk 1]
```diff
@@ -157,11 +157,18 @@ export const _TagsSurvey = (props: _TagsSurveyProps): JSX.Element => {
...
             if (prevState[issueCategory]?.issues) {
-              (prevState[issueCategory]!.issues as unknown[]) = prevState[issueCategory]!.issues!.filter(function (
-                value
-              ) {
-                return value !== issue;
-              });
+              // 'prevState[issueCategory]!.issues as ...' typing is required here to avoid a typescript limitation
+              // "This expression is not callable" caused by filter().
+              // More information can be found here: https://github.com/microsoft/TypeScript/issues/44373
+              (prevState[issueCategory]!.issues as (_AudioIssue | _OverallIssue | _ScreenshareIssue | _VideoIssue)[]) =
```
Comment (emlynmac) [inline packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: Can this be cleaned up at all? 
Ultimately, the filter is is if the value is equal to the issue - is there a way to make this simpler?

Comment (edwardlee-msft) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: This is not an easy unfortunately. More info can be found in this github typescript issue if you would like more details regarding it https://github.com/microsoft/TypeScript/issues/36390.

One way to fix it is through typing arrays like this:
something: (object | object)[]. 
But in this case, prevState is already a _CallSurvey type. No further typing can be enhanced here. If you have any other suggestions please let me know, I'll be more than happy to try it out.

Comment (emlynmac) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: Ok, that's a fun long-running TS issue....

**Review by emlynmac — APPROVED** [review]
Let's keep an eye on the upstream TS issue and if that gets resolved, update ours.

File: packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx (hunk 1 above)
Comment (JamesBurnside) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: Out of interest, how do this only appear in this PR and not in the react-components PR?

Comment (JamesBurnside) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: Oh it was previously `as unknown`, this is just slightly stronger - nice!

Comment (JamesBurnside) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: In setStates, it is not best practice to mutate the prevState and return the mutated state. You are better to create a new object and return that.

I'm struggling to fully understand this snippet, but is this the same as:
```ts
setSelectedTags((prevState) => {
  const categoryIssues = prevState[issueCategory]?.issues?.filter((value) => value !== issue);
  return {
    ...prevState,
    [issueCategory]: { issues: categoryIssues }
  };
});
```
? If it is that would simplify signifanctly

Comment (JamesBurnside) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: But honestly I find this whole code block very hard to parse so that could be wrong

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx (hunk 1 above)
Comment (edwardlee-msft) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx:163]: With this snippet code we still hit the "filter" issue as mentioned in the previous comments. So an as typing is still required. I'll continue to do more testing with surveyTags to see if the snippet code causes any changes to the functionality.

Comment (edwardlee-msft) [reply packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx]: Tested on Calling sample and is functioning correctly :D
